src/s_sound.c

```diff
@@ -463,6 +370,16 @@ void S_ResumeSound(void)
 // Updates music & sounds
 //
 
+void S_InitListener(const mobj_t *listener)
+{
+    if (nosfxparm)
+        return;
+
+    I_DeferSoundUpdates();
+    I_UpdateListenerParams(listener);
+    I_ProcessSoundUpdates();
+}
+
 void S_UpdateSounds(const mobj_t *listener)
```

Comment:
The last few commits were made to (1) restore the channel assignment behavior in `S_StartSound()`, and (2) reduce redundant "listener" calculations each time `S_UpdateSounds()` is called.

For (1), I realized I had inadvertently skipped calls to `S_getChannel()` in the event that all sound channels were being used because I had moved channel assignment earlier in `S_StartSound()`. I don't want to change that behavior at all. This required a bit of refactoring.

For (2), I noticed that `S_UpdateSounds()` is called [as frequently as possible](https://github.com/fabiangreffrath/woof/blob/06939bd9a6501e99ae55e27a5879df6512d10d33/src/d_main.c#L2741-L2757), meaning 35 times a second when the FPS is capped, or thousands of times a second when uncapped. The listener is only updated once per frame now, but an additional optimization can be made by _always_ limiting the sound updates to 35 Hz. Using an independent audio update rate is a recommended practice that is explained in the [effects extension guide](https://openal-soft.org/misc-downloads/Effects%20Extension%20Guide.pdf) (pages 44-45). I'll take a look at this optimization next.

Reply:
> an additional optimization can be made by always limiting the sound updates to 35 Hz

Here's an idea: https://github.com/fabiangreffrath/woof/pull/1118/commits/83512b69d5faf19c4ecb6ed668c67930f5ff83e2

Reply:
> Here's an idea: [83512b6](https://github.com/fabiangreffrath/woof/commit/83512b69d5faf19c4ecb6ed668c67930f5ff83e2)

Good enough for me. 

> Using an independent audio update rate is a recommended practice that is explained in the [effects extension guide](https://openal-soft.org/misc-downloads/Effects%20Extension%20Guide.pdf) (pages 44-45). I'll take a look at this optimization next.

Will this be part of this PR?

Reply:
> Will this be part of this PR?

That's taken care of by the same change: https://github.com/fabiangreffrath/woof/commit/83512b69d5faf19c4ecb6ed668c67930f5ff83e2